app/controllers/blog_posts_controller.rb

```diff
@@ -18,10 +19,18 @@ def show
   private
 
     def blog_post_params
-      params.permit(:tag, :slug)
+      params.permit(:slug)
+    end
+
+    def blog_tag_params
+      params.permit(:tag)
     end
 
     def find_blog_post!
       @blog_post = BlogPost.unscoped_find_by!(blog_post_params)
     end
+
+    def check_tag_exists!
```

Comment:
the method name should be in line with the other one, what it does is it finds the tag. so I'd call it `find_tag` (just like `find_blog_post`)